FEM-2796 Added more details to LoadError and GeneralError

GouravSna opened the pull request:
- Added message to GeneralError and LoadError for PhoenixMediaProvider and KalturaOvpMediaProvider
- Removed redundant interface method from AndroidTest classes.

GouravSna committed:
- Added message to GeneralError and LoadError for PhoenixMediaProvider and KalturaOvpMediaProvider
- Removed redundant interface method from AndroidTest classes.

GouravSna committed:
- Code refactor

giladna commented on mediaproviders/src/main/java/com/kaltura/playkit/providers/ovp/KalturaOvpMediaProvider.java:
```diff
@@ -337,7 +337,7 @@ private void onEntryInfoMultiResponse(String ks, ResponseElement response, OnMed
 
             } else {
-                error = response.getError() != null ? response.getError() : ErrorElement.LoadError;
+                error = response.getError() != null ? response.getError() : ErrorElement.LoadError.message("error response in multirequest. response: " + response.getResponse());
```
@GouravSna 
please verify if response is not null before using the getter

GouravSna replied:
@giladna , This check in line no 288, in isValidResponse() method.

GouravSna committed:
Merge branch 'develop' into FEM-2796

GouravSna committed:
- Creating new object of ErrorElement
- Removed static ErrorElement function calls
- Static import / import refactor

giladna committed:
add

    public static ErrorElement buildGeneralErrorElement(String message) {
        return new ErrorElement(ErrorElement.GeneralError.getName(), message, ErrorElement.ErrorCode.GeneralErrorCode);
    }

    public static ErrorElement buildLoadErrorElement(String message) {
        return new ErrorElement(ErrorElement.LoadError.getName(), message,ErrorElement.ErrorCode.LoadErrorCode);
    }

    public static ErrorElement buildNotFoundlErrorElement(String message) {
        return new ErrorElement(ErrorElement.NotFound.getName(), message, ErrorElement.ErrorCode.NotFoundCode);
    }

    public static ErrorElement buildBadRequestErrorElement(String message) {
        return new ErrorElement(ErrorElement.BadRequestError.getName(), message, ErrorElement.ErrorCode.BadRequestErrorCode);
    }

giladna committed:
add

    public static ErrorElement buildGeneralErrorElement(String message) {
        return new ErrorElement(ErrorElement.GeneralError.getName(), message, ErrorElement.ErrorCode.GeneralErrorCode);
    }

    public static ErrorElement buildLoadErrorElement(String message) {
        return new ErrorElement(ErrorElement.LoadError.getName(), message,ErrorElement.ErrorCode.LoadErrorCode);
    }

    public static ErrorElement buildNotFoundlErrorElement(String message) {
        return new ErrorElement(ErrorElement.NotFound.getName(), message, ErrorElement.ErrorCode.NotFoundCode);
    }

    public static ErrorElement buildBadRequestErrorElement(String message) {
        return new ErrorElement(ErrorElement.BadRequestError.getName(), message, ErrorElement.ErrorCode.BadRequestErrorCode);
    }

giladna committed:
cont

giladna committed:
use static import
